pkg/compact/compact.go

```diff
@@ -140,9 +143,15 @@ func NewSyncer(logger log.Logger, reg prometheus.Registerer, bkt objstore.Bucket
 		bkt:                  bkt,
 		metrics:              newSyncerMetrics(reg),
 		blockSyncConcurrency: blockSyncConcurrency,
+		indexCacher:          NewNoopIndexCacheProcessor(),
```

Comment:
it should be always set, no noop needed.

Reply:
It's weird to do composability for a plugin that is *only temporary* as we mentioned earlier. I would rather see it behind some single flag like `generateMissingIndexCacheFiles` bool